### PR TITLE
Remove ability to update planning application created_at date

### PR DESCRIPTION
**Pull request by benbaumann95**
### Description of change

Remove ability to update planning application created_at date
- Add received_at column and allow users to update this . We should never permit a created_at timestamp to be updated

### Story Link

https://trello.com/c/agAgh2He/1079-high-priority-remove-ability-to-update-createdat-timestamp-on-planning-applications

### CODE REVIEW

File: app/models/planning_application.rb [hunk 1]
```diff
@@ -303,7 +303,7 @@ def documents_for_decision_notice
   end
 
   def received_at
-    Time.next_immediate_business_day(created_at)
+    super || (Time.next_immediate_business_day(created_at) if created_at)
```
Comment (allbecauseyoutoldmeso) [inline app/models/planning_application.rb:306]: I think `created_at` should still always be present?

**Review by allbecauseyoutoldmeso — APPROVED** [review]
Looks good!

Just wondering if there's any value to retrospectively updating `received_at` for existing records?

File: app/models/planning_application.rb (hunk 1 above)
Comment (benbaumann95) [reply app/models/planning_application.rb:306]: Yeah left it out but then on planning application create page it was calling this code `Time.next_immediate_business_day(created_at)` and nil was blowing up. Maybe I can edit that [method](https://github.com/unboxed/bops/blob/main/config/initializers/business_time.rb#L13-L15) directly though just to return if the argument is nil 

What do you reckon?

**Comment by benbaumann95** [comment]
> Looks good!
> 
> Just wondering if there's any value to retrospectively updating `received_at` for existing records?

My understanding was that it's just an override if the next business day code isn't quite correct.
I'm not sure what an accurate update would look like 🤔

File: app/models/planning_application.rb (hunk 1 above)
Comment (allbecauseyoutoldmeso) [reply app/models/planning_application.rb:306]: Ah, I hadn't thought of new records! I think it's probably best as you've already done it, with the check in the model.

**Pull request merged**